File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:

    def isAnagram(self, s: str, t: str) -> bool:
        

        if len(t) != len(s):
            return False

        maHash = {}

        for char in s:
            maHash[char] = maHash.get(char, 0) + 1
        for char in t:
            maHash[char] = maHash.get(char, 0) - 1

        return all(count == 0 for count in maHash.values())

        return False

File: 0242-valid-anagram/0242-valid-anagram.cs
public class Solution {
    public bool IsAnagram(string s, string t) {

        if(s.Length != t.Length)
            return false;

        var maHash = s.GroupBy(c => c).ToDictionary(d => d.Key, d => d.Count());

        foreach (var a in t)
        {
            if(!maHash.TryGetValue(a, out var count) || count ==0)
                return false;
            maHash[a]--;
        }
        return true;
    }
}